test/performance/range/views/view_translate_1D_benchmark.cpp

```diff
@@ -0,0 +1,170 @@
+    #include <seqan/seq_io.h>
+    #include <seqan/translation.h>
+#endif
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct baseline_tag{};
+struct translate_tag{};
+
+// ============================================================================
+//  sequential_read
+// ============================================================================
+
+template <typename rng_t>
+void sequential_read_impl(benchmark::State & state, rng_t && rng)
+{
+    for (auto _ : state)
+    {
+        for (aa27 c : rng)
+        {
+            benchmark::DoNotOptimize(c.to_rank());
+        }
+    }
+}
+
+template <typename tag_t>
+void sequential_read(benchmark::State & state)
+{
+    std::vector<dna4> c{test::generate_sequence<seqan3::dna4>(1000, 0, 0)};
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<aa27> cpy = c | views::translate_single | views::to<std::vector<aa27>>;
+        sequential_read_impl(state, cpy);
+    }
+    else if constexpr (std::is_same_v<tag_t, translate_tag>)
+    {
+        auto v = c | views::translate_single;
+        sequential_read_impl(state, v);
+    }
+}
+
+BENCHMARK_TEMPLATE(sequential_read, baseline_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_tag);
+
+// ============================================================================
+//  random_access
+// ============================================================================
+
+template <typename rng_t>
+void random_access_impl(benchmark::State & state, rng_t && rng, std::vector<size_t> const & access_positions)
+{
+    for (auto _ : state)
+    {
+        for (auto pos : access_positions)
+        {
+            benchmark::DoNotOptimize(rng[pos].to_rank());
+        }
+    }
+}
+
+template <typename tag_t>
+void random_access(benchmark::State & state)
+{
+    std::vector<dna4> c{test::generate_sequence<seqan3::dna4>(10000, 0, 0)};
+
+    std::vector<size_t> access_positions;
+    access_positions.resize(200);
+    std::mt19937 gen(42);
+    std::uniform_int_distribution<size_t> dis_position(0, 1000 - 1);
+
+    for (size_t i = 0; i < 200; ++i)
+        access_positions[i] = dis_position(gen);
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<aa27> cpy = c | views::translate_single | views::to<std::vector<aa27>>;
+        random_access_impl(state, cpy, access_positions);
+    }
+    else
+    {
+        auto v = c | views::translate_single;
+        random_access_impl(state, v, access_positions);
+    }
+}
+
+BENCHMARK_TEMPLATE(random_access, baseline_tag);
+BENCHMARK_TEMPLATE(random_access, translate_tag);
+
+// ============================================================================
+//  copy_vector
+// ============================================================================
+
+template <typename adaptor_t>
+void copy_impl(benchmark::State & state, std::vector<dna4> const & c, adaptor_t & adaptor)
+{
+    for (auto _ : state)
+    {
+        std::vector<aa27> cpy;
```

Comment:
Could you explain to me why?

Reply:
The problem before c++11 was that `std::vector<aa27> cpy()` would not do what you would expect. This syntax declares a function with name `cpy` that returns a `std::vector`, so you had to leave the parentheses away. However, most of the other high-level languages don't do it like this and especially since all other constructors need to be called with parenthesis this seems akward. So in order to make it expressive that you want in deed a value initialisation of the object you call it with empty braces. Moreover, if the underlying type is an aggregate you will initialise it with aggregate initialisation, which is almost always what you want. So we explicitly stated in our style guide to use braces to initialise objects.